### ISSUE TITLE
Unactionable "failed to find bin" cargo dist build error (wrong path?)

### ISSUE THREAD

**Issue by HadrienG2** [axodotdev/cargo-dist#362]
So, I have this project which I recently updated from cargo dist v0.0.5 to v0.1: https://github.com/HadrienG2/crofiler .

If I run `cargo dist build` locally (or in CI, which is where I found it first), I get the infamous "failed to find bin" error.

```text
analyzing workspace:
  simpp
    [bin] simpp
  cpparser (no binaries)
  asylum (no binaries)
  crofiler
    [bin] crofiler
  clang-time-trace (no binaries)
  cmakeperf (dist = false)
    [bin] mock

bundling simpp-x86_64-unknown-linux-gnu.tar.xz
bundling simpp-x86_64-unknown-linux-gnu.tar.xz.sha256
bundling simpp-installer.sh
bundling crofiler-x86_64-unknown-linux-gnu.tar.xz
bundling crofiler-x86_64-unknown-linux-gnu.tar.xz.sha256
bundling crofiler-installer.sh
building cargo target (x86_64-unknown-linux-gnu/dist --workspace)
   Compiling crofiler v2.0.0 (/home/hadrien/Bureau/crofiler)
    Finished dist [optimized + debuginfo] target(s) in 10.13s
  × failed to find bin simpp (simpp 2.0.0 (path+file:///home/hadrien/Bureau/crofiler/simpp))
```

However, I do not understand why this error happens since a `cargo build --profile dist --workspace` will seemingly correctly produce all the binaries I want:

```text
$ l target/dist/
total 49556
drwxr-xr-x 1 hadrien users      442 25 août  08:41 ./
drwxr-xr-x 1 hadrien users      142 25 août  08:29 ../
drwxr-xr-x 1 hadrien users     3774 25 août  08:32 build/
-rw-r--r-- 1 hadrien users        0 25 août  08:29 .cargo-lock
-rwxr-xr-x 2 hadrien users 26159504 25 août  08:41 crofiler*
-rw-r--r-- 1 hadrien users     4063 25 août  08:32 crofiler.d
drwxr-xr-x 1 hadrien users    29470 25 août  08:41 deps/
drwxr-xr-x 1 hadrien users        0 25 août  08:29 examples/
drwxr-xr-x 1 hadrien users    12158 25 août  08:41 .fingerprint/
drwxr-xr-x 1 hadrien users        0 25 août  08:29 incremental/
-rw-r--r-- 1 hadrien users      208 25 août  08:41 libasylum.d
-rw-r--r-- 2 hadrien users   250992 25 août  08:32 libasylum.rlib
-rw-r--r-- 1 hadrien users     2595 25 août  08:41 libclang_time_trace.d
-rw-r--r-- 2 hadrien users  6295790 25 août  08:32 libclang_time_trace.rlib
-rw-r--r-- 1 hadrien users      403 25 août  08:41 libcmakeperf.d
-rw-r--r-- 2 hadrien users  2245050 25 août  08:32 libcmakeperf.rlib
-rw-r--r-- 1 hadrien users     1782 25 août  08:41 libcpparser.d
-rw-r--r-- 2 hadrien users  5132764 25 août  08:32 libcpparser.rlib
-rwxr-xr-x 2 hadrien users  2989448 25 août  08:41 mock*
-rw-r--r-- 1 hadrien users      446 25 août  08:41 mock.d
-rwxr-xr-x 2 hadrien users  7623512 25 août  08:41 simpp*
-rw-r--r-- 1 hadrien users     1819 25 août  08:41 simpp.d
```

The only possible clue I have is that the path in the error message does not look correct: it sounds like it's looking for a file in `/home/hadrien/Bureau/crofiler/simpp` when it should be looking into `/home/hadrien/Bureau/crofiler/target/dist/simpp` (notice the missing `target/dist` path components).

How can I go about debugging this further?

**Comment by HadrienG2** [comment]
PS : In addition to cargo-dist v0.1, I have tested that this error is still present on the current git master.

**Comment by Gankra** [comment]
Hey, sorry about that! It appears that cargo_metadata 0.15.0 was silently failing to parse the compiler messages (dumping them all to Message::TextLine) so cargo-dist thought none of the binaries were getting built.

Updating to 0.17.0 seems to have fixed it (#363).

**Comment by HadrienG2** [comment]
That the cargo-metadata v0.17 helps makes sense : it [adds support for named debuginfo levels](https://github.com/oli-obk/cargo_metadata/pull/241) and I took advantage of these for faster builds in my latest MSRV version bump (which is what the release I was tagging was about to begin with).

OTOH, that's not a nice way for a parser to fail a parse :frowning_face: . Shame on cargo_metadata, and good catch on your side!

I can confirm that this fixes the problem on my side too.
